cpp/src/arrow/compute/api_scalar.h

```diff
@@ -52,6 +53,13 @@ class ARROW_EXPORT ElementWiseAggregateOptions : public FunctionOptions {
   bool skip_nulls;
 };
 
+/// Functor to calculate hash of an enum.
```

Comment:
nit: maybe this should go in arrow/util, e.g. arrow/util/map.h or (maybe possibly) arrow/util/hashing.h?

Reply:
Will place in `arrow/util/map.h`